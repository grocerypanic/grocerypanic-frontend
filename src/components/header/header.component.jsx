import React from "react";

import StopIcon from "@material-ui/icons/Stop";
import AddIcon from "@material-ui/icons/Add";
import HomeIcon from "@material-ui/icons/Home";
import StoreIcon from "@material-ui/icons/Store";
import KitchenIcon from "@material-ui/icons/Kitchen";

import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";

import { Navbar, Nav, Spinner } from "react-bootstrap";
import {
  NavContainer,
  IconContainer,
  OffIconContainer,
  MenuContainer,
} from "./header.styles";

import Routes from "../../configuration/routes";

const Header = ({ history, title, create, transaction }) => {
  React.useEffect(() => {}, []);

  const navigate = (route) => {
    if (transaction) return;
    history.push(route);
  };

  return (
    <NavContainer collapseOnSelect variant="dark" fixed="top">
      <Navbar.Brand href="#home">Panic: {title}</Navbar.Brand>
      <Nav className="mr-auto"></Nav>
      <Nav>
        <Nav.Item className="nav-link" onClick={() => navigate(Routes.root)}>
          <MenuContainer
            route={Routes.root}
            history={history}
            data-testid="home-icon"
          >
            <HomeIcon />
          </MenuContainer>
        </Nav.Item>
      </Nav>
      <Nav>
        <Nav.Item className="nav-link" onClick={() => navigate(Routes.stores)}>
          <MenuContainer
            route={Routes.stores}
            history={history}
            data-testid="stores-icon"
          >
            <StoreIcon />
          </MenuContainer>
        </Nav.Item>
      </Nav>
      <Nav>
        <Nav.Item className="nav-link" onClick={() => navigate(Routes.shelves)}>
          <MenuContainer
            route={Routes.shelves}
            history={history}
            data-testid="shelves-icon"
          >
            <KitchenIcon />
          </MenuContainer>
        </Nav.Item>
      </Nav>
      <Nav>
        {create ? (
          <Nav.Item className="nav-link" onClick={create}>
            <IconContainer>
              {transaction ? (
                <Spinner size="sm" animation="border" />
              ) : (
                <AddIcon className="AddIcon" />
              )}
            </IconContainer>
          </Nav.Item>
        ) : (
          <Nav.Item className="nav-link">
            <OffIconContainer>
              <StopIcon />
            </OffIconContainer>
          </Nav.Item>
        )}
      </Nav>
    </NavContainer>
  );
};

export default withRouter(Header);

Header.propTypes = {
  create: PropTypes.func,
  title: PropTypes.string.isRequired,
  transaction: PropTypes.bool.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};
