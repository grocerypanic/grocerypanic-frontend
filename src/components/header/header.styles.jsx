import { Navbar } from "react-bootstrap";
import styled, { css } from "styled-components";
import { primary } from "../../configuration/theme";
import { white, navbarSize } from "../../configuration/theme";

const container = css`
  height: ${navbarSize};
  width: ${navbarSize};
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const NavContainer = styled(Navbar)`
  background-color: ${primary};
  height: ${navbarSize};
  padding: 2px;
  padding-left: 10px;

  .action {
    cursor: pointer;
  }
  .fit {
    padding-left: 3px !important;
    padding-right: 3px !important;
  }
  .header-hidden {
    display: none;
  }
`;

export const OffIconContainer = styled.div`
  ${container}
  color: ${primary};
  background-color: ${primary};
`;

export const IconContainer = styled.div`
  ${container}
  background-color: ${primary};
`;

export const MenuContainer = styled.div`
  ${(props) =>
    props.route === props.history.location.pathname
      ? `color: ${white};`
      : null};
`;
