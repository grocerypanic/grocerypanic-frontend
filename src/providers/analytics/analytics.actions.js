export const AnalyticsActions = {
  TestAction: { category: "Test", action: "Submitted a test action" },
  LoginError: { category: "Error", action: "Unable to login with backend" },
};
