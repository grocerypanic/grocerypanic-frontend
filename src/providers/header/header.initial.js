const InitialValues = {
  headerSettings: {
    title: "",
    create: null,
    transaction: false,
    disableNav: false,
    signIn: false,
  },
  updateHeader: null,
};

export default InitialValues;
