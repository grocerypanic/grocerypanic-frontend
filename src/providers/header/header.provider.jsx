import React, { useState, createContext } from "react";
import InitialState from "./header.initial";

export const HeaderContext = createContext({ ...InitialState });

const HeaderProvider = ({ children }) => {
  const [headerSettings, setHeaderSettings] = useState(
    InitialState.headerSettings
  );

  const updateHeader = (newSettings) => {
    setHeaderSettings(newSettings);
  };

  return (
    <HeaderContext.Provider
      value={{
        headerSettings,
        updateHeader,
      }}
    >
      {children}
    </HeaderContext.Provider>
  );
};

export default HeaderProvider;
