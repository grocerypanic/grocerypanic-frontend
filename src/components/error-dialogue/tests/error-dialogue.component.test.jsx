import WarningOutlinedIcon from "@material-ui/icons/WarningOutlined";
import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { createBrowserHistory } from "history";
import i18next from "i18next";
import React from "react";
import { Router } from "react-router-dom";
import Routes from "../../../configuration/routes";
import Strings from "../../../configuration/strings";
import { AnalyticsActions } from "../../../providers/analytics/analytics.actions";
import { AnalyticsContext } from "../../../providers/analytics/analytics.provider";
import initialHeaderSettings from "../../../providers/header/header.initial";
import { HeaderContext } from "../../../providers/header/header.provider";
import ErrorDialogue from "../error-dialogue.component";

jest.mock("@material-ui/icons/WarningOutlined", () => ({
  __esModule: true,
  default: jest.fn(),
}));
WarningOutlinedIcon.mockImplementation(() => <div>MockWarning</div>);

const mockHeaderUpdate = jest.fn();
const mockEvent = jest.fn();
const mockAnalyticsSettings = { event: mockEvent, initialized: true };

let ErrorProps = {
  clearError: jest.fn(),
  eventMessage: AnalyticsActions.TestAction,
  messageTranslationKey: "Testing.GenericMultiLineTranslationTestString",
  redirect: Routes.menu,
};

const history = createBrowserHistory();

const renderHelper = (history, currentProps) => {
  return render(
    <HeaderContext.Provider
      value={{ ...initialHeaderSettings, updateHeader: mockHeaderUpdate }}
    >
      <Router history={history}>
        <AnalyticsContext.Provider value={mockAnalyticsSettings}>
          <ErrorDialogue {...currentProps} history={history} />
        </AnalyticsContext.Provider>
      </Router>
    </HeaderContext.Provider>
  );
};

describe("Setup Environment", () => {
  let utils;
  let current;
  let testRoute;

  beforeEach(() => {
    jest.clearAllMocks();
    testRoute = "/test/route";
    current = { ...ErrorProps };
    history.push(Routes.menu);
  });

  describe("when given a valid analytics event message", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      current.eventMessage = AnalyticsActions.TestAction;
    });

    describe("when given a redirect to go back", () => {
      beforeEach(() => {
        current.redirect = Routes.goBack;
        history.push(testRoute);
        utils = renderHelper(history, { ...current });
      });

      it("renders, should call header with the correct params", () => {
        expect(mockHeaderUpdate).toHaveBeenCalledTimes(1);
        expect(mockHeaderUpdate).toBeCalledWith({
          title: "ErrorDialogue.ErrorDialogueHeaderTitle",
          create: null,
          transaction: false,
          disableNav: true,
        });
        expect(i18next.t("ErrorDialogue.ErrorDialogueHeaderTitle")).toBe(
          Strings.ErrorDialogue.ErrorDialogueHeaderTitle
        );
      });

      it("should render with the correct components and messages", () => {
        expect(
          utils.getAllByText(Strings.Testing.GenericTranslationTestString)
            .length
        ).toBe(2);
        expect(WarningOutlinedIcon).toHaveBeenCalledTimes(1);
        expect(utils.getByTestId("ErrorConfirmation")).toBeTruthy();
      });

      it("should send the event as expected", () => {
        expect(mockEvent).toHaveBeenCalledTimes(1);
        expect(mockEvent).toHaveBeenCalledWith(AnalyticsActions.TestAction);
      });

      it("should handle the confirmation button as expected, route goes back", async () => {
        const button = utils.getByTestId("ErrorConfirmation");
        expect(ErrorProps.clearError).toHaveBeenCalledTimes(0);
        fireEvent.click(button);
        expect(ErrorProps.clearError).toHaveBeenCalledTimes(1);
        await waitFor(() =>
          expect(history.location.pathname).not.toBe(testRoute)
        );
        await waitFor(() =>
          expect(history.location.pathname).toBe(Routes.menu)
        );
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });

    describe("when given a route with no redirect", () => {
      beforeEach(() => {
        delete current.redirect;
        history.push(testRoute);
        utils = renderHelper(history, { ...current });
      });

      it("renders, should call header with the correct params", () => {
        expect(mockHeaderUpdate).toHaveBeenCalledTimes(1);
        expect(mockHeaderUpdate).toBeCalledWith({
          title: "ErrorDialogue.ErrorDialogueHeaderTitle",
          create: null,
          transaction: false,
          disableNav: true,
        });
        expect(i18next.t("ErrorDialogue.ErrorDialogueHeaderTitle")).toBe(
          Strings.ErrorDialogue.ErrorDialogueHeaderTitle
        );
      });

      it("should render with the correct components and messages", () => {
        expect(
          utils.getAllByText(Strings.Testing.GenericTranslationTestString)
            .length
        ).toBe(2);
        expect(WarningOutlinedIcon).toHaveBeenCalledTimes(1);
        expect(mockEvent).toHaveBeenCalledTimes(1);
        expect(mockEvent).toHaveBeenCalledWith(AnalyticsActions.TestAction);
        expect(utils.getByTestId("ErrorConfirmation")).toBeTruthy();
      });

      it("should handle the confirmation button as expected, route goes back", async () => {
        const button = utils.getByTestId("ErrorConfirmation");
        expect(ErrorProps.clearError).toHaveBeenCalledTimes(0);
        fireEvent.click(button);
        expect(ErrorProps.clearError).toHaveBeenCalledTimes(1);
        await waitFor(() => expect(history.location.pathname).toBe(testRoute));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });

    describe("when given a redirect", () => {
      beforeEach(() => {
        current.redirect = "/brand/new/route";
        history.push(testRoute);
        utils = renderHelper(history, { ...current });
      });

      it("renders, should call header with the correct params", () => {
        expect(mockHeaderUpdate).toHaveBeenCalledTimes(1);
        expect(mockHeaderUpdate).toBeCalledWith({
          title: "ErrorDialogue.ErrorDialogueHeaderTitle",
          create: null,
          transaction: false,
          disableNav: true,
        });
        expect(i18next.t("ErrorDialogue.ErrorDialogueHeaderTitle")).toBe(
          Strings.ErrorDialogue.ErrorDialogueHeaderTitle
        );
      });

      it("should render with the correct components and messages", () => {
        expect(
          utils.getAllByText(Strings.Testing.GenericTranslationTestString)
            .length
        ).toBe(2);
        expect(WarningOutlinedIcon).toHaveBeenCalledTimes(1);
        expect(mockEvent).toHaveBeenCalledTimes(1);
        expect(mockEvent).toHaveBeenCalledWith(AnalyticsActions.TestAction);
        expect(utils.getByTestId("ErrorConfirmation")).toBeTruthy();
      });

      it("should handle the confirmation button as expected, route goes back", async () => {
        const button = utils.getByTestId("ErrorConfirmation");
        expect(ErrorProps.clearError).toHaveBeenCalledTimes(0);
        fireEvent.click(button);
        expect(ErrorProps.clearError).toHaveBeenCalledTimes(1);
        await waitFor(() =>
          expect(history.location.pathname).toBe(current.redirect)
        );
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });
  });

  describe("when given an invalid analytics event message", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      current.eventMessage = null;
    });

    describe("when given any redirect type", () => {
      beforeEach(() => {
        current.redirect = Routes.goBack;
        history.push(testRoute);
        utils = renderHelper(history, { ...current });
      });

      it("renders, should call header with the correct params", () => {
        expect(mockHeaderUpdate).toHaveBeenCalledTimes(1);
        expect(mockHeaderUpdate).toBeCalledWith({
          title: "ErrorDialogue.ErrorDialogueHeaderTitle",
          create: null,
          transaction: false,
          disableNav: true,
        });
        expect(i18next.t("ErrorDialogue.ErrorDialogueHeaderTitle")).toBe(
          Strings.ErrorDialogue.ErrorDialogueHeaderTitle
        );
      });

      it("should render with the correct components and messages", () => {
        expect(
          utils.getAllByText(Strings.Testing.GenericTranslationTestString)
            .length
        ).toBe(2);
        expect(WarningOutlinedIcon).toHaveBeenCalledTimes(1);
        expect(utils.getByTestId("ErrorConfirmation")).toBeTruthy();
      });

      it("should not send any the event", () => {
        expect(mockEvent).toHaveBeenCalledTimes(0);
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });
  });

  afterEach(cleanup);
});
