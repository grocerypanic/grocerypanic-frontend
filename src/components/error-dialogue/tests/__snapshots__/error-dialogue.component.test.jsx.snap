// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment when given a valid analytics event message when given a redirect should match the snapshot on file (styles) 1`] = `
.c3 {
  padding-left: 0px;
  margin: 0px;
  min-width: 200px;
  background: #808080;
  padding: 10px;
}

.c4 {
  margin: 0px;
  min-width: 200px;
  padding: 10px;
}

.c6 {
  margin-top: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #FF0000;
  color: #FFFFFF;
  text-align: center;
  overflow: hidden;
}

.c5 {
  text-align: center;
}

.c0 {
  margin-top: calc(50px + 26px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div>
        MockWarning
      </div>
    </div>
    <h1>
      Error!
    </h1>
    <ul
      class="c3"
    >
      <div
        class="c4 alert alert-danger"
      >
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
      </div>
    </ul>
    <button
      class="c6 btn btn-success"
      data-testid="ErrorConfirmation"
    >
      OK
    </button>
  </div>
</div>
`;

exports[`Setup Environment when given a valid analytics event message when given a redirect to go back should match the snapshot on file (styles) 1`] = `
.c3 {
  padding-left: 0px;
  margin: 0px;
  min-width: 200px;
  background: #808080;
  padding: 10px;
}

.c4 {
  margin: 0px;
  min-width: 200px;
  padding: 10px;
}

.c6 {
  margin-top: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #FF0000;
  color: #FFFFFF;
  text-align: center;
  overflow: hidden;
}

.c5 {
  text-align: center;
}

.c0 {
  margin-top: calc(50px + 26px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div>
        MockWarning
      </div>
    </div>
    <h1>
      Error!
    </h1>
    <ul
      class="c3"
    >
      <div
        class="c4 alert alert-danger"
      >
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
      </div>
    </ul>
    <button
      class="c6 btn btn-success"
      data-testid="ErrorConfirmation"
    >
      OK
    </button>
  </div>
</div>
`;

exports[`Setup Environment when given a valid analytics event message when given a route with no redirect should match the snapshot on file (styles) 1`] = `
.c3 {
  padding-left: 0px;
  margin: 0px;
  min-width: 200px;
  background: #808080;
  padding: 10px;
}

.c4 {
  margin: 0px;
  min-width: 200px;
  padding: 10px;
}

.c6 {
  margin-top: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #FF0000;
  color: #FFFFFF;
  text-align: center;
  overflow: hidden;
}

.c5 {
  text-align: center;
}

.c0 {
  margin-top: calc(50px + 26px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div>
        MockWarning
      </div>
    </div>
    <h1>
      Error!
    </h1>
    <ul
      class="c3"
    >
      <div
        class="c4 alert alert-danger"
      >
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
      </div>
    </ul>
    <button
      class="c6 btn btn-success"
      data-testid="ErrorConfirmation"
    >
      OK
    </button>
  </div>
</div>
`;

exports[`Setup Environment when given an invalid analytics event message when given any redirect type should match the snapshot on file (styles) 1`] = `
.c3 {
  padding-left: 0px;
  margin: 0px;
  min-width: 200px;
  background: #808080;
  padding: 10px;
}

.c4 {
  margin: 0px;
  min-width: 200px;
  padding: 10px;
}

.c6 {
  margin-top: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #FF0000;
  color: #FFFFFF;
  text-align: center;
  overflow: hidden;
}

.c5 {
  text-align: center;
}

.c0 {
  margin-top: calc(50px + 26px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div>
        MockWarning
      </div>
    </div>
    <h1>
      Error!
    </h1>
    <ul
      class="c3"
    >
      <div
        class="c4 alert alert-danger"
      >
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
        <div
          class="c5"
        >
          GenericTranslationTestString
        </div>
      </div>
    </ul>
    <button
      class="c6 btn btn-success"
      data-testid="ErrorConfirmation"
    >
      OK
    </button>
  </div>
</div>
`;
